dotcom-rendering/src/lib/decideLogo.ts

```diff
@@ -1,35 +1,51 @@
-import { ArticleDesign, ArticleSpecial, Pillar } from '@guardian/libs';
+import { ArticleDesign } from '@guardian/libs';
 import type { Branding } from '../types/branding';
+import type { DCRContainerPalette } from '../types/front';
 
-const shouldUseLogoForDarkBackground = (format: ArticleFormat): boolean => {
-	if (format.theme === ArticleSpecial.SpecialReport) return true;
-	switch (format.design) {
-		case ArticleDesign.Gallery:
-		case ArticleDesign.Audio:
-		case ArticleDesign.Video:
+const shouldUseLogoForDarkBackground = (
+	format: ArticleFormat,
+	containerPalette?: DCRContainerPalette,
+): boolean => {
+	switch (containerPalette) {
+		// Special palettes with dark background colours on containers
+		case 'BreakingPalette':
+		case 'SombrePalette':
+		case 'InvestigationPalette':
+		case 'SombreAltPalette':
 			return true;
-		case ArticleDesign.LiveBlog:
-			switch (format.theme) {
-				case ArticleSpecial.Labs:
-					return false;
-				case Pillar.News:
-				case Pillar.Sport:
-				case Pillar.Opinion:
-				case Pillar.Lifestyle:
-				case Pillar.Culture:
-				default:
+
+		// Special palettes with light background colours on containers
+		case 'Branded':
+		case 'EventAltPalette':
+		case 'EventPalette':
+		case 'LongRunningAltPalette':
+		case 'LongRunningPalette':
+		case 'SpecialReportAltPalette':
+			return false;
+
+		// Special palettes which act more like standard containers
+		case 'MediaPalette':
+		case 'PodcastPalette':
+		case undefined:
+		default: {
```

Comment:
Do you need the `default` case here? It’s safer to be exhaustive, [as demonstrated in `typescript-school`](https://github.com/guardian/typescript-school/blob/main/4-narrowing/slides.md#default-cases) and [following this thread in chat](https://chat.google.com/room/AAAAWwBdSMs/fKTo8f-UfG4/DZMBSQ5_AJQ?cls=10).